python/mp/rosary.py

```diff
@@ -220,42 +286,83 @@ def add_effect_object(self, effect):
         self.effects.append(effect)
         return self.effect_id
 
-    #def add_effect(self, name, bead_set, color=color.Color(1,1,1)):
-    #def add_effect(self, name, bead_set_name, color=color.Color(1,1,1)):
-    @dm.expose()
-    def add_effect(self, name, bead_set_name='all', color_name_or_r=None,
-                   g=None, b=None, duration=None):
-        """Adds an Effect to the active Effect list by using the Effect
-        name. Returns the id of the active effect.
 
+    def add_trigger_object(self, trigger):
+        """
+        When we "fire" a trigger, create a new trigger object like we do for
+        effects, then run through its `inner_fire()`, and promptly exit
         """
-        print("ADD THIS EFFECT: {}".format(name))
+        self.trigger_id += 1
+        trigger.id = self.trigger_id
+        trigger.rosary = self
+        self.triggers.append(trigger)
+        trigger.fire()
+        return self.trigger_id
+
+
+    @dm.expose()
+    def add_effect(self, *args, **kwargs):
```

Comment:
*args here is basically unused, but it is because of the **kwargs here that we can be as flexible with initializing effects from the OSC endpoints as we can in python.